bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/connection/ConnectionManager.java

```diff
@@ -0,0 +1,530 @@
+                            return;
+                        }
+                        if (bodyType != -64) {
+                            if (bodyType == 30) {
+                                logger.warn("Error response 0x1E received {} from IP Address:{}", bodyType, ipAddress);
+                                return;
+                            }
+                            logger.warn("Unexpected response bodyType {}", bodyType);
+                            return;
+                        }
+                        lastResponse = new Response(data, version, "", bodyType);
+                        try {
+                            logger.trace("Data length is {}, version is {}, Ip Address is {}", data.length, version,
+                                    ipAddress);
+                            if (callback != null) {
+                                callback.updateChannels(lastResponse);
+                            }
+                        } catch (Exception ex) {
+                            logger.warn("Processing response exception: {}", ex.getMessage());
+                        }
+                    }
+                }
+                return;
+            } else {
+                if (retry) {
+                    logger.debug("Resending Command {}", command);
+                    retry = false;
+                    sendCommand(command, callback);
+                } else {
+                    droppedCommands = droppedCommands + 1;
+                    logger.info("Problem with reading response, skipping {} skipped count since startup {}", command,
+                            droppedCommands);
+                    retry = true;
+                    return;
+                }
+            }
+        } catch (SocketException e) {
+            droppedCommands = droppedCommands + 1;
+            logger.debug("Socket exception on IP: {}, skipping command {} skipped count since startup {}", ipAddress,
+                    command, droppedCommands);
+            throw new MideaConnectionException(e);
+        } catch (IOException e) {
+            droppedCommands = droppedCommands + 1;
+            logger.debug("IO exception on IP: {}, skipping command {} skipped count since startup {}", ipAddress,
+                    command, droppedCommands);
+            throw new MideaConnectionException(e);
+        }
+    }
+
+    /**
+     * Closes all elements of the connection before starting a new one
+     * Makes sure writer, inputStream and socket are closed before each command is started
+     */
+    public synchronized void disconnect() {
+        logger.debug("Disconnecting from device at {}", ipAddress);
+
+        InputStream inputStream = this.inputStream;
+        DataOutputStream writer = this.writer;
+        Socket socket = this.socket;
+        try {
+            writer.close();
+            inputStream.close();
+            socket.close();
+
```

Comment:
```suggestion
```

Reply:
I think I have all these addressed in next commit